src/plugins/image/editing/image-editor.js

```diff
@@ -36,79 +41,76 @@ function ImageEditor({ content, onContentChanged, publicStorage, privateStorage
   useEffect(() => {
     const src = currentImageSource;
     if (!src) {
-      setShouldWarnOfSmallImageSize(false);
+      setSmallImageSizeWarning(null);
+      return;
+    }
+    if (effectType === EFFECT_TYPE.clip) {
       return;
     }
 
     (async () => {
-      const dimensions = await getImageDimensions(src);
-      if (src === currentImageSource) {
-        setShouldWarnOfSmallImageSize(dimensions && dimensions.width < SMALL_IMAGE_WIDTH_THRESHOLD);
+      if (src !== currentImageSource) {
+        return;
       }
+      const dimensions = await getImageDimensions(src);
+      setSmallImageSizeWarning(dimensions && dimensions.width < SMALL_IMAGE_WIDTH_THRESHOLD ? t('smallImageSizeWarning') : null);
     })();
-  }, [currentImageSource]);
+  }, [currentImageSource, effectType, t]);
+
+  useEffect(() => {
+    if (!currentImageSource) {
+      setSmallImageSizeWarning(null);
+      return;
+    }
+    if (effectType !== EFFECT_TYPE.clip) {
+      return;
+    }
+
+    const img = clipEffectImageRef.current;
+    const clipWidth = Math.round(img.naturalWidth * (effect.region.width / 100));
+    const clipHeight = Math.round(img.naturalHeight * (effect.region.height / 100));
+
+    setClippingSizeInPx({ width: clipWidth, height: clipHeight });
+    setSmallImageSizeWarning(clipWidth < SMALL_IMAGE_WIDTH_THRESHOLD ? t('smallClippingSizeWarning') : null);
+  }, [currentImageSource, effectType, effect?.region, t]);
```

Comment:
Could we also put `clipEffectImageRef.current` here?

Reply:
Actually while doing this I tested a bit more and discovered some strange inconsistencies when switching images and reworked the logic a bit to fix the issues.

Reply:
OK, was on the wrong track anyways. Refs cannot be dependencies.